### ISSUE TITLE
Connect timeout with TemporaryProfile

### ISSUE THREAD

**Issue by DircoDT** [jorgebv/windows-wifi-api#6]
I am connecting to a network using the `Wlan.WlanConnectionMode.TemporaryProfile` mode (because I don't want to add/modify existing profiles.)

```
ConnectSynchronously(Wlan.WlanConnectionMode.TemporaryProfile, Wlan.Dot11BssType.Any, profileString, timeoutInMilliseconds);
```

Normally the `profileString` input parameter above would only contain the profile name, but for temporary profiles it seems you have to send in that whole XML string.

This results in a timeout because the following check will always be false, even if it connects successfully:

```
    switch ((Wlan.WlanNotificationCodeAcm)wlanConnectionData.notifyData.notificationCode)
    {
        case Wlan.WlanNotificationCodeAcm.ConnectionComplete:
            if (wlanConnectionData.connNotifyData.profileName == profile)
```

That last comparison is always false, because the one side is just a name and the other is the full XML string.

To fix it, you could probably do a different check if using TemporaryProfile, or just add a check for the XML contents as well, e.g.

```
            if ((wlanConnectionData.connNotifyData.profileName == profile) ||
                (profile.Contains("<name>"+ wlanConnectionData.connNotifyData.profileName + "</name>")))
```

**Comment by jorgebv** [comment]
I haven't tested this but I do believe you are right. I will repro this and fix it soon.

**Comment by DircoDT** [comment]
For testing, I use an Android phone as a wi-fi hotspot, with the default settings (WPA2PSK, etc.)
Here is the profile xml string for it, if you want to try something similar:

```
    const string profileXmlTemplate = "<?xml version=\"1.0\"?><WLANProfile xmlns=\"http://www.microsoft.com/networking/WLAN/profile/v1\"><name>{0}_TEMP</name><SSIDConfig><SSID><name>{0}</name></SSID></SSIDConfig><connectionType>ESS</connectionType><connectionMode>manual</connectionMode><autoSwitch>false</autoSwitch><MSM><security><authEncryption><authentication>WPA2PSK</authentication><encryption>AES</encryption><useOneX>false</useOneX></authEncryption><sharedKey><keyType>passPhrase</keyType><protected>false</protected><keyMaterial>{1}</keyMaterial></sharedKey></security></MSM></WLANProfile>";

    const string mySsid = "AndroidAP";
    const string myKey = "some_plain_text_unencrypted_key";

    string myTestProfile = string.Format(profileXmlTemplate, mySsid, myKey);
```

**Comment by jorgebv** [comment]
Thanks for the xml. This should be fixed now by 5dda3aabd8a8a30f298535b43267981909fae2d9
